azerothcore/azerothcore-wotlk
# fix(Core/BG/PvPstats): store correct winner_faction

## Changes Proposed:
- Add enum from TC
- Fix value of field `winner_faction` in `pvpstats_battleground` table 
see https://www.azerothcore.org/wiki/pvpstats_battlegrounds#winnerfaction

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10308
- Closes https://github.com/chromiecraft/chromiecraft/issues/2842

## Tests Performed:

- Builds


## How to Test the Changes:


1. Set `Battleground.StoreStatistics.Enable` to `1`
2. Play a BG
3. Check that the last new row created inside the `pvpstats_battleground` table of the `acore_characters` DB has the correct `winner_faction` value
4. 
![image](https://user-images.githubusercontent.com/75517/152896976-84bf7b0b-9e82-45d9-9bb3-17a99ea0a286.png)



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
